tests/bfd/test_bfd_static_route.py

```diff
@@ -0,0 +1,1369 @@
+            for each_member in list_of_portchannel_members_on_src:
+                self.control_interface_state(src_dut, src_asic, bfd_base_instance, each_member, action)
+        
+        # Verification of BFD session state.
+        assert wait_until(300, 10, 0, lambda: bfd_base_instance.verify_bfd_state(dst_dut, dst_dut_nexthops.values(), dst_asic, "Up"))
+        assert wait_until(300, 10, 0, lambda: bfd_base_instance.verify_bfd_state(src_dut, src_dut_nexthops.values(), src_asic, "Up"))
+
+        # Verify that corresponding static route has been added on both duts
+        logger.info("Static route verifications")
+        self.verify_static_route(request, dst_asic, dst_prefix, dst_dut, dst_dut_nexthops, "Route Addition", bfd_base_instance, version)
+        self.verify_static_route(request, src_asic, src_prefix, src_dut, src_dut_nexthops, "Route Addition", bfd_base_instance, version)
+
+    def test_bfd_with_rp_config_reload_ipv4(self, localhost, duthost, request, duthosts, tbinfo, get_src_dst_asic_and_duts, bfd_base_instance, enum_supervisor_dut_hostname, bfd_cleanup_db):
+        version = "ipv4"
+
+        rp = duthosts[enum_supervisor_dut_hostname]
+
+        # Selecting source, destination dut & prefix & BFD status verification for all nexthops
+        logger.info("Selecting Source dut, destination dut, source asic, destination asic, source prefix, destination prefix")
+        src_asic, dst_asic, src_dut, dst_dut, src_dut_nexthops, dst_dut_nexthops, src_prefix, dst_prefix = self.select_src_dst_dut_with_asic(request, get_src_dst_asic_and_duts, bfd_base_instance, version)
+        
+        # Creation of BFD
+        logger.info("BFD addition on source dut")   
+        bfd_base_instance.add_bfd(src_asic.asic_index, src_prefix, src_dut)
+        
+        logger.info("BFD addition on destination dut")   
+        bfd_base_instance.add_bfd(dst_asic.asic_index, dst_prefix, dst_dut)
+        
+        # Verification of BFD session state.
+        assert wait_until(180, 10, 0, lambda: bfd_base_instance.verify_bfd_state(dst_dut, dst_dut_nexthops.values(), dst_asic, "Up"))
+        assert wait_until(180, 10, 0, lambda: bfd_base_instance.verify_bfd_state(src_dut, src_dut_nexthops.values(), src_asic, "Up"))
+        
+        # Savings the configs
+        src_dut.shell('sudo config save -y')
+        dst_dut.shell('sudo config save -y')
+        
+        # Perform a cold reboot on source dut
+        config_reload(rp)
+
+        # Waiting for all processes on Source & destination dut
+        wait_critical_processes(src_dut)
+        wait_critical_processes(dst_dut)
+
+        check_bgp = request.getfixturevalue("check_bgp")
+        results = check_bgp()
+        failed = [result for result in results if "failed" in result and result["failed"]]
+        if failed:
+            pytest.fail("BGP check failed, not all BGP sessions are up. Failed: {}".format(failed))
+
```

Comment:
Transition verification required.
Add log analyzer regex to ensure DOWN string is observed in syslogs.